lib/accesslists/hierarchy.go

```diff
@@ -323,78 +330,59 @@ func IsAccessListOwner(
 	return accesslistv1.AccessListUserAssignmentType_ACCESS_LIST_USER_ASSIGNMENT_TYPE_UNSPECIFIED, trace.Wrap(ownershipErr)
 }
 
-// IsAccessListMember checks if the given user is the Access List member. It returns an error
-// matched by [IsUserLocked] if the user is locked.
+// IsAccessListMember validates that:
+// - the user is not locked
+// - the user is directly or indirectly member of the given access list
+// - the user is meeting the memberships requirements
+// The function returns the type of assignment, unspecified means the user is
+// not a member of the access list. If the user is not member of an accesslist,
+// an explanation can be returned.
+// The returned error only indicates that something unexpected happened when
+// resolving the accesslist memberships. If everything went correctly but the
+// user is not a member, the error will be nil.
 func IsAccessListMember(
 	ctx context.Context,
 	user types.User,
 	accessList *accesslist.AccessList,
 	g AccessListAndMembersGetter,
 	lockGetter services.LockGetter,
 	clock clockwork.Clock,
-) (accesslistv1.AccessListUserAssignmentType, error) {
+) (accesslistv1.AccessListUserAssignmentType, string, error) {
```

Comment:
Does returning a 'reason` is  really required  ? 

The `IsAccessListMember` is not user facing function  and the `reason` complicates the flow. 

I don't have strong opinion if we decided a reson we need to extend the docs to make sure how to use this function. 

For simplicity personally i would just do following or remove the reason entirely:

```
func IsAccessListMember(...) {
    res, _, err := IsAccessListMemberWithReason(....)
    return res, trace.Wrap(err)
}
```

If you want to preserve the reason please update the docs and describe why the `reason` is needed